File: backend/src/main/java/vn/edu/fpt/doghandbook/backend/service/HealthRecordService.java
package vn.edu.fpt.doghandbook.backend.service;

import vn.edu.fpt.doghandbook.backend.dto.request.HealthRecordRequest;
import vn.edu.fpt.doghandbook.backend.dto.response.HealthRecordResponse;
import vn.edu.fpt.doghandbook.backend.dto.response.PageResponse;

public interface HealthRecordService {

    PageResponse<HealthRecordResponse> getAll(int page, int size);

    PageResponse<HealthRecordResponse> getByDog(Integer dogId, int page, int size);

    HealthRecordResponse getById(Integer recordId);

    HealthRecordResponse create(HealthRecordRequest request, Integer examinerId);

    HealthRecordResponse update(Integer recordId, HealthRecordRequest request, Integer examinerId);
}

File: web-admin/src/services/healthRecordService.js
import api from './api';

export const healthRecordService = {
  getByDog: (dogId, page = 0, size = 10) =>
    api.get(`/health-records/by-dog/${dogId}?page=${page}&size=${size}`),
  getById: (id) => api.get(`/health-records/${id}`),
  create: (data) => api.post('/health-records', data),
  update: (id, data) => api.put(`/health-records/${id}`, data),
};
